ReadMe.md

```diff
@@ -0,0 +1,360 @@
+| 200           | Success       |
+| 401           | Unauthorized  |
+| 403           | Forbidden     |
+| 404           | Not Found     |
+
+### Product
+### Cart
+User is filling up cart with requested products.
+### Order
+### User
+User can register and set up own account. Each account is secured with a password.
+#### ![#00FF00](https://placehold.it/15/00FF00/000000?text=+) GET
+*Method used to get all users. No parameters required.*
+#####*Model*
+- address (string, optional),
+- blocked (boolean, optional),
+- id (integer, optional),
+- name (string, optional),
+- password (string, optional)
+#####*Request URL*
+```bash
+http://localhost:8080/v1/user
+```
+#####*curl*
+```bash
+curl -X GET --header 'Accept: application/json' 'http://localhost:8080/v1/user'
+```
+#####*Example response*
+```bash
+[
+  {
+     "id": 1,
+        "name": "user1",
+        "password": "$2aonhK4V5.",
+        "blocked": false,
+        "address": "address1"
+  }
+]
+```
+#####*Response Messages*
+| Code          | Explanation   |
+| ------------- | ------------- |
+| 200           | Success       |
+| 401           | Unauthorized  |
+| 403           | Forbidden     |
+| 404           | Not Found     |
+
+#### ![#00FF00](https://placehold.it/15/00FF00/000000?text=+) GET
+*Method used to get chosen user. Parameter required.*
+#####*Parameter type*
+- Long
+#####*Request URL*
+```bash
+http://localhost:8080/v1/user/{id}
+```
+#####*curl*
+```bash
+curl -X GET --header 'Accept: application/json' 'http://localhost:8080/v1/user/{id}'
+```
+#####*Example response*
+```bash
+  {
+    "id": 1,
+      "name": "user1",
+      "password": "$XonhK4V5.",
+      "blocked": false,
+      "address": "address1"
+  }
+```
+#####*Response Messages*
+| Code          | Explanation   |
+| ------------- | ------------- |
+| 200           | Success       |
+| 401           | Unauthorized  |
+| 403           | Forbidden     |
+| 404           | Not Found     |
+
+####![#c5f015](https://placehold.it/15/c5f015/000000?text=+) POST
+*Method used to create new user. Parameter required*
+#####*Parameter content type*
+- json
+
+#####*Request URL*
+```bash
+http://localhost:8080/v1/user
+```
+#####*curl*
+```bash
+curl -X POST --header 'Content-Type: application/json' --header 'Accept: */*' -d '{ \ 
+   "address": "string", \ 
+   "blocked": true, \ 
+   "id": 0, \ 
+   "name": "string", \ 
+   "password": "string" \ 
+ }' 'http://localhost:8080/v1/user'
+```
+#####*Example request body*
+```bash
+ {
+   "address": "string",
+     "blocked": true,
+     "id": 0,
+     "name": "string",
+     "password": "string"
+ }
+```
+#####*Response Messages*
+| Code          | Explanation   |
+| ------------- | ------------- |
+| 200           | Success       |
+| 401           | Unauthorized  |
+| 403           | Forbidden     |
+| 404           | Not Found     |
+
+####![#1589F0](https://placehold.it/15/1589F0/000000?text=+) PUT
+*Method used to update user data. Parameter required.*
```

Comment:
W tym przypadku należy podkreślić, że metoda PUT nie zmieni nazwy uzytkownika! Jeśli wyślemy PUT z nazwą inną, niż jest w bazie, to wyrzuci błąd